src/state/ducks/posts/selectors.js

```diff
@@ -0,0 +1,37 @@
+const sort = (posts, sortBy, source) => {
+  if (source && posts[source] !== undefined) {
+    return sortBy === 'new' ?
+      posts[source].sort(post => new Date(post.timestamp) - new Date()) :
+      posts[source].sort((postF, postS) =>
+        (postS.meta.numViewed + postS.meta.numShared + postS.meta.numSaved) -
+        (postF.meta.numViewed + postF.meta.numShared + postF.meta.numSaved));
+  }
+  return sortBy === 'new' ?
+    posts.sort(post => new Date(post.timestamp) - new Date()) :
+    posts.sort((postF, postS) =>
+      (postS.meta.numViewed + postS.meta.numShared + postS.meta.numSaved) -
+      (postF.meta.numViewed + postF.meta.numShared + postF.meta.numSaved));
+};
+
+const getHomePosts = state => sortBy => (
+  sort(Object.values(state.posts.home), sortBy)
+);
+
+const getForYouPosts = state => sortBy => (
+  sort(Object.values(state.posts.forYou), sortBy)
+);
+
+const getSavedPosts = state => sortBy => (
+  sort(Object.values(state.posts.saved), sortBy)
+);
+
+const getSourcePosts = state => (sortBy, source) => (
```

Comment:
the `source` param is required but the `sortBy` is optional, so we should put the source to the front